lib/eliscore/user.ex

```diff
@@ -22,17 +23,32 @@ defmodule Eliscore.User do
   def changeset(%User{} = user, attrs) do
     user
     |> cast(attrs, @required_fields, @optional_fields)
-    |> validate_required([:login, :email,:password])
+    |> validate_required([:login, :email])
     |> validate_format(:email, ~r/@/)
-    |> generate_encrypted_password 
+    |> generate_encrypted_password
+  end
+
+  def find_or_create(%Auth{} = auth) do
```

Comment:
first of all take it from model to service like `Eliscore.Services.Auth` and put there something like this:
```def sign_in_user(conn, changeset) do
    case insert_or_update(changeset) do
      {:ok, user} ->
        Guardian.encode_and_sign(user, :token)

        auth_conn = Guardian.Plug.api_sign_in(conn, user)
        jwt = Guardian.Plug.current_token(auth_conn)
        Guardian.Plug.claims(auth_conn)

        auth_conn
        |> put_resp_header("authorization", "Bearer #{jwt}")
        |> json(%{access_token: jwt})
      {:error, _changeset} -> 
        conn
        |> put_status(401)
        |> render(MyAppWeb.ErrorView, "401.json-api")
    end
  end

  defp insert_or_update(changeset) do
    case Repo.get_by(User, email: changeset.changes.email) do
      nil -> 
        changeset
        |> force_change(:provider, "google")
        |> Repo.insert!
      user -> {:ok, user}
    end
  end
```
You will have the idea what I am talking about 🤔